### PR TITLE
Adds and use the component to display picked files during a new transfer

### PR DESCRIPTION
**Pull request by LunarX**
For now it displays picked files without copying them to the app's local storage. This will come in a second PR

Depends on #85

### CODE REVIEW

File: app/src/main/res/values-fr/strings.xml [hunk 1]
```diff
@@ -60,6 +64,11 @@
         <item quantity="one">%d jour</item>
         <item quantity="other">%d jours</item>
     </plurals>
+    <string name="transferSendButton">Envoyer</string>
+    <plurals name="transferSpaceLeft">
```
Comment (KevinBoulongne) [inline app/src/main/res/values-fr/strings.xml:68]:
```suggestion
    <plurals name="transferSpaceLeft" tools:ignore="MissingQuantity">
```

File: app/src/main/res/values-fr/strings.xml [hunk 2]
```diff
@@ -22,6 +22,10 @@
     <string name="contentDescriptionButtonClose">Fermer</string>
     <string name="contentDescriptionButtonRemove">Supprimer le fichier</string>
     <string name="contentDescriptionCreateNewTransferButton">Nouveau transfert</string>
+    <plurals name="filesCount">
+        <item quantity="one">%d fichier</item>
+        <item quantity="other">%d fichiers</item>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-fr/strings.xml:27]: `quantity="many"` is missing.

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import kotlinx.parcelize.Parcelize
+
+@Composable
+fun SelectedFilesCard(
+    modifier: Modifier = Modifier,
+    files: () -> List<FileUiItem>,
+    formattedSizeWithUnits: () -> String,
+    showUploadSourceChoiceBottomSheet: () -> Unit,
+    removeFileByUid: (uid: String) -> Unit
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:56]:
```suggestion
    removeFileByUid: (uid: String) -> Unit,
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+            items(
+                items = files().asReversed(),
+                key = { TransferLazyRowKey(TransferLazyRowKey.Type.FILE, it.uid) },
+            ) { file ->
+                SmallFileTile(
+                    modifier = Modifier.animateItem(),
+                    file = file,
+                    smallFileTileSize = SmallFileTileSize.LARGE,
+                    onRemove = { removeFileByUid(file.uid) }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:105]:
```suggestion
                    onRemove = { removeFileByUid(file.uid) },
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                formatSpaceLeft(formattedSizeWithUnits),
+                color = SwissTransferTheme.colors.secondaryTextColor,
+                style = SwissTransferTheme.typography.bodySmallRegular,
+            )
+            Spacer(modifier = Modifier.weight(1f))
+            Icon(
+                imageVector = AppImages.AppIcons.ChevronRightSmall,
+                contentDescription = null,
+                modifier = Modifier.padding(Margin.Medium),
+                tint = SwissTransferTheme.colors.iconColor
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:84]:
```suggestion
                tint = SwissTransferTheme.colors.iconColor,
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            )
+        }
+
+        LazyRow(
+            Modifier.fillMaxWidth(),
+            contentPadding = PaddingValues(start = Margin.Medium, end = Margin.Medium, bottom = Margin.Medium),
+            horizontalArrangement = Arrangement.spacedBy(Margin.Medium)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:91]:
```suggestion
            horizontalArrangement = Arrangement.spacedBy(Margin.Medium),
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+            imageVector = AppImages.AppIcons.AddThick,
+            contentDescription = null,
+        )
+    }
+}
+
+@Parcelize
+private data class TransferLazyRowKey(
+    val type: Type,
+    val fileUid: String? = null
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:153]:
```suggestion
    val fileUid: String? = null,
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferCard.kt [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import com.infomaniak.swisstransfer.ui.theme.Shapes
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+
+@Composable
+fun SwissTransferCard(modifier: Modifier = Modifier, content: @Composable ColumnScope.() -> Unit) {
+    Card(
+        modifier = modifier,
+        shape = Shapes.medium,
+        colors = CardDefaults.cardColors(contentColor = SwissTransferTheme.colors.secondaryTextColor),
+        content = content
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/SwissTransferCard.kt:43]:
```suggestion
        content = content,
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import com.infomaniak.library.filetypes.FileType
+import com.infomaniak.swisstransfer.ui.theme.LocalIsDarkMode
+import com.infomaniak.swisstransfer.ui.utils.fileType
+import com.infomaniak.swisstransfer.ui.utils.hasPreview
+
+@Composable
+fun FilePreview(
+    file: FileUiItem,
+    circleColor: Color,
+    circleSize: Dp
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt:47]:
```suggestion
    circleSize: Dp,
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
+        Canvas(modifier = Modifier.size(circleSize)) {
+            drawCircle(color = color)
+        }
+
+        Icon(
+            modifier = Modifier.size(circleSize / 2),
+            imageVector = fileType.icon,
+            contentDescription = null,
+            tint = fileType.color(LocalIsDarkMode.current)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt:83]:
```suggestion
            tint = fileType.color(LocalIsDarkMode.current),
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/CrossCircleButton.kt [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+            contentPadding = PaddingValues(0.dp),
+            shape = CircleShape,
+            colors = ButtonDefaults.buttonColors(containerColor = SwissTransferTheme.colors.fileItemRemoveButtonBackground),
+            onClick = onClick ?: {},
+        ) {
+            Icon(
+                modifier = Modifier.size(Margin.Small),
+                imageVector = AppImages.AppIcons.CrossThick,
+                contentDescription = stringResource(R.string.contentDescriptionButtonRemove),
+                tint = Color.White
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/CrossCircleButton.kt:60]:
```suggestion
                tint = Color.White,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt [hunk 12]
```diff
@@ -17,47 +17,53 @@
...
+    removeFileByUid: (uid: String) -> Unit,
     addFiles: (List<Uri>) -> Unit,
-    navigateToTransferTypeScreen: () -> Unit,
     closeActivity: () -> Unit,
 ) {
+    val context = LocalContext.current
     var showUploadSourceChoiceBottomSheet by rememberSaveable { mutableStateOf(true) }
-    val isNextButtonEnabled by remember { derivedStateOf { transferFiles().isNotEmpty() } }
+    val formattedSizeWithUnits by remember {
+        derivedStateOf {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt:61]: It's useless, use only `remember` with files list as key

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt [hunk 13]
```diff
@@ -114,11 +106,26 @@ private fun ImportFilesScreen(
     )
 }
 
+private fun getFormattedSizeWithUnits(usedSpace: Long, context: Context): String {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt:109]:
```suggestion
private fun getHumanReadableSize(usedSpace: Long, context: Context): String {
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+
+@Composable
+fun SelectedFilesCard(
+    modifier: Modifier = Modifier,
+    files: () -> List<FileUiItem>,
+    formattedSizeWithUnits: () -> String,
+    showUploadSourceChoiceBottomSheet: () -> Unit,
+    removeFileByUid: (uid: String) -> Unit,
+) {
+    val fileCount by remember { derivedStateOf { files().count() } }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:58]:
```suggestion
    val fileCount = files().count()
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+    val quantity = LocalContext.current.getQuantityFromFormattedSizeWithUnits(formattedSize)
+    return pluralStringResource(R.plurals.transferSpaceLeft, quantity, formattedSize)
+}
+
+private fun Context.getQuantityFromFormattedSizeWithUnits(formattedSize: String): Int {
+    val sizeParts = formattedSize.split(' ', Typography.nbsp) // Space for languages such as EN and NBSP for languages such as FR
+
+    return if (sizeParts.size == 2) {
+        val local = resources.configuration.getLocales().get(0)
+        val parsedNumber = NumberFormat.getInstance(local).parse(sizeParts[0])
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:124]: replace get(0) by `first`

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+                    smallFileTileSize = SmallFileTileSize.LARGE,
+                    onRemove = { removeFileByUid(file.uid) },
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun formatSpaceLeft(formattedSizeWithUnits: () -> String): String {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt:113]: Extract to own utils file

**Review by sirambd — CHANGES_REQUESTED** [review]

**Comment by sirambd** [comment]
Fix Sonar issues

File: app/src/main/res/values-fr/strings.xml [hunk 17]
```diff
@@ -22,6 +22,10 @@
     <string name="contentDescriptionButtonClose">Fermer</string>
     <string name="contentDescriptionButtonRemove">Supprimer le fichier</string>
     <string name="contentDescriptionCreateNewTransferButton">Nouveau transfert</string>
+    <plurals name="filesCount">
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/res/values-fr/strings.xml:28]: No, we want to add the `many` here.

[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/res/values-fr/strings.xml:28]: I think we want the `quantity="many"` for this one.

File: app/src/main/res/values-fr/strings.xml [hunk 18]
```diff
@@ -70,6 +74,11 @@
         <item quantity="one">%d jour</item>
         <item quantity="other">%d jours</item>
     </plurals>
+    <string name="transferSendButton">Envoyer</string>
+    <plurals name="transferSpaceLeft">
```
Comment (KevinBoulongne) [inline app/src/main/res/values-fr/strings.xml:78]: I think we want the `quantity="many"` for this one.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-fr/strings.xml (hunk 17 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/res/values-fr/strings.xml:28]: I think we want the `quantity="many"` for this one.

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+        )
+    }
+}
```
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/components/FilePreview.kt:86]: Knowing all the refactor there's been about previews and FileUiItem right on top of this PR, maybe it'd be wiser to add the preview in a new PR

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt (hunk 15 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/SelectedFilesCard.kt]: LocalList has not .first() extension unfortunately

**Review by sirambd — APPROVED** [review]

**Review by tevincent — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**